src/Avalonia.Controls/Primitives/TextSelectionHandle.cs

```diff
@@ -0,0 +1,210 @@
+﻿using System;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.Media;
+
+namespace Avalonia.Controls.Primitives
+{
+    public class TextSelectionHandle : TemplatedControl
+    {
+        internal static readonly RoutedEvent<VectorEventArgs> DragStartedEvent =
+            RoutedEvent.Register<TextSelectionHandle, VectorEventArgs>(nameof(DragStarted), RoutingStrategies.Bubble);
+
+        internal static readonly RoutedEvent<VectorEventArgs> DragDeltaEvent =
+            RoutedEvent.Register<TextSelectionHandle, VectorEventArgs>(nameof(DragDelta), RoutingStrategies.Bubble);
+
+        internal static readonly RoutedEvent<VectorEventArgs> DragCompletedEvent =
+            RoutedEvent.Register<TextSelectionHandle, VectorEventArgs>(nameof(DragCompleted), RoutingStrategies.Bubble);
+
+        internal static readonly StyledProperty<SelectionHandleType> SelectionHandleTypeProperty = AvaloniaProperty.Register<TextSelectionHandle, SelectionHandleType>(nameof(SelectionHandleType));
+
+        internal event EventHandler<VectorEventArgs> DragStarted
+        {
+            add { AddHandler(DragStartedEvent, value); }
+            remove { RemoveHandler(DragStartedEvent, value); }
+        }
+
+        internal event EventHandler<VectorEventArgs> DragDelta
+        {
+            add { AddHandler(DragDeltaEvent, value); }
+            remove { RemoveHandler(DragDeltaEvent, value); }
+        }
+
+        internal event EventHandler<VectorEventArgs> DragCompleted
+        {
+            add { AddHandler(DragCompletedEvent, value); }
+            remove { RemoveHandler(DragCompletedEvent, value); }
+        }
+
+        internal SelectionHandleType SelectionHandleType
```

Comment:
The type `SelectionHandleType` is public, but this property isn't. Shouldn't it? Otherwise there's no real point of it being a `StyledProperty`.